### PR TITLE
Deployment test image tag added

### PR DESCRIPTION
**Pull request by Madhan-SWE**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
Testing the deployment of release image.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: deploy/kubernetes/overlays/stable/kustomization.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ bases:
   - ../../base
 images:
   - name: gcr.io/k8s-staging-cloud-provider-ibm/ibm-powervs-block-csi-driver
-    newTag: main
+    newTag: test-image-v1
```
Comment (mkumatag) [inline deploy/kubernetes/overlays/stable/kustomization.yaml:7]: lets add next alpha version to be realistic,

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: deploy/kubernetes/overlays/stable/kustomization.yaml (hunk 1 above)
Comment (Madhan-SWE) [reply deploy/kubernetes/overlays/stable/kustomization.yaml]: fixed

File: README.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ Please see the compatibility matrix above before you deploy the driver
 
 To deploy the CSI driver:
 ```sh
-kubectl apply -k "https://github.com/kubernetes-sigs/ibm-powervs-block-csi-driver/deploy/kubernetes/overlays/stable"
+kubectl apply -k "https://github.com/kubernetes-sigs/ibm-powervs-block-csi-driver/deploy/kubernetes/overlays/stable/?ref=v0.1.0-alpha.2"
```
Comment (mkumatag) [inline README.md:76]: This doesn't work as it already released and will pick older version image, we should rename this to `v0.1.0-alpha.3` and then cut the release immediately and then try pulling.

File: deploy/kubernetes/overlays/stable/kustomization.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ bases:
   - ../../base
 images:
   - name: gcr.io/k8s-staging-cloud-provider-ibm/ibm-powervs-block-csi-driver
-    newTag: main
+    newTag: v0.1.0-alpha.2
```
Comment (mkumatag) [inline deploy/kubernetes/overlays/stable/kustomization.yaml:7]: I'm thinking of having 2 overlays like:
for staging `overlays/staging` for the prereleases
for production `overlays/stable` pointing to production images `gcr.io/cloud-provider-ibm/ibm-powervs-block-csi-driver`

**Review by mkumatag — CHANGES_REQUESTED** [review]
Lets discuss on this in detail if needed.

File: deploy/kubernetes/overlays/sidecars/main_image.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  - name: k8s.gcr.io/sig-storage/csi-provisioner
+    newTag: v3.1.0
+  - name: k8s.gcr.io/sig-storage/csi-attacher
+    newTag: v3.4.0
+  - name: k8s.gcr.io/sig-storage/csi-resizer
+    newTag: v1.4.0
+  - name: k8s.gcr.io/sig-storage/livenessprobe
+    newTag: v2.6.0
+  - name: k8s.gcr.io/sig-storage/csi-node-driver-registrar
+    newTag: v2.5.0
```
Comment (mkumatag) [inline deploy/kubernetes/overlays/sidecars/main_image.yaml:16]: newline needed

File: deploy/kubernetes/overlays/sidecars/main_image.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+bases:
+  - ../../base
+  -
```
Comment (mkumatag) [inline deploy/kubernetes/overlays/sidecars/main_image.yaml:5]: why this empty entry here?

**Review by mkumatag — CHANGES_REQUESTED** [review]
We also need an entry for the staging repository folder right?

File: deploy/kubernetes/overlays/sidecars/main_image.yaml (hunk 4 above)
Comment (Madhan-SWE) [reply deploy/kubernetes/overlays/sidecars/main_image.yaml]: outdated.

File: deploy/kubernetes/overlays/sidecars/main_image.yaml (hunk 5 above)
Comment (Madhan-SWE) [reply deploy/kubernetes/overlays/sidecars/main_image.yaml]: outdated.

File: README.md (hunk 2 above)
Comment (Madhan-SWE) [reply README.md]: updated.

File: deploy/kubernetes/overlays/stable/kustomization.yaml (hunk 3 above)
Comment (Madhan-SWE) [reply deploy/kubernetes/overlays/stable/kustomization.yaml]: Concluded that 2 folders not needed.

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**